OrbitGl/App.cpp

```diff
@@ -873,6 +873,7 @@ void OrbitApp::OnProcessSelected(int32_t pid) {
   std::shared_ptr<Process> process = FindProcessByPid(pid);
   CHECK(process != nullptr);
   Capture::SetTargetProcess(std::move(process));
+  FireRefreshCallbacks();
```

Comment:
Could this be moved below UpdateModuleList(pid); ?

Reply:
This is actually already in UpdateModuleList(pid) at the very end. However, this method spawns a task on another thread (from the main thread) which then again spawns a task on the main thread. In this last task the FireRefreshCallbacks is called at the very end. And the UI seems to redraw the FunctionsView before that call arrives, and thus has inconsistent data (new process (including functions) and old list of indices of functions to draw), which finally leads to the crash.

I am not sure if there is a better way to solve this without blocking the UI for longer time.
Steps to reproduce and further info are in the bug (b/163786513). I would leave this open for now if you want to look a bit further into this and will merge that change for now.

Reply:
I am pretty indifferent between this line as a fix or dimitrys idea in the comment below. I think both should work. If you go with this, maybe `FireRefreshCallbacks(DataViewType::FUNCTIONS);` is enough?

Reply:
I also tried this, but I am not sure, if there were similar bugs in the other views (even though there shouldn't). Anyways, went for the other solution.